core/arch/arm/mm/core_mmu.c

```diff
@@ -643,7 +649,8 @@ bool core_pbuf_is(uint32_t attr, paddr_t pbuf, size_t len)
 	case CORE_MEM_SEC:
 		return pbuf_is_inside(secure_only, pbuf, len);
 	case CORE_MEM_NON_SEC:
-		return pbuf_is_inside(nsec_shared, pbuf, len);
+		return pbuf_is_inside(nsec_shared, pbuf, len) ||
+			pbuf_is_inside(ddr, pbuf, len);
```

Comment:
This is the place where DDR layout should be checked.
Verifying in DDR is a good start but I think we must reject any SEC_RAM mapped DDR memories (if some platform declares in the DDR) and exclude also the SDP (bootime) registered memories (`pbuf_is_sdp_mem()`). And there are cases where core is inside DDR. This is about preventing core to faulty access a secure memory with non-secure attribute.

For this, we will need some devs. sorry. Maybe, before allowing SHM registration at boot time, we could allocate a table to list the DDR fragments eligible to dynamic SHM registration.